fontforge/svg.c

```diff
@@ -2043,7 +2043,7 @@ static void xmlParseColorSource(xmlNodePtr top,char *name,DBounds *bbox,
     *_grad = NULL; *_epat = NULL;
     if ( colour_source==NULL )
 	LogError(_("Could not find Color Source with id %s."), name );
-    else if ( (islinear = xmlStrcmp(colour_source->name,(xmlChar *) "linearGradient")==0) ||
+    else if ( ((islinear = xmlStrcmp(colour_source->name,(xmlChar *) "linearGradient"))==0) ||
```

Comment:
If xmlStrcmp is like strcmp, then the placement of brackets here is now wrong - a zero return value from strcmp indicates a match, so comparing its return value `== 0` would indicate the presence of a linear gradient. So you probably want something like 

````c
else if ((islinear = (xmlStrcmp(colour_source->name, (xmlChar *) "linearGradient") == 0))
````